tests/test_slim_engine.py

```diff
@@ -194,6 +194,7 @@ def test_simulate(self):
         self.assertEqual(observed_counts[2], 8)
         self.assertTrue(all(tree.num_roots == 1 for tree in ts.trees()))
 
+    @unittest.skip("Test broken: something wrong with popen mock?")
```

Comment:
Not sure what happened here @grahamgower - I couldn't figure out why this test was after breaking.

Reply:
Looking at it now...

Reply:
There is something funny going on with the mock.patch, and I'm not sure what that is. I guess autospeccing here is insufficient. In any event, the mocked `subprocess.Popen` is also used internally by `subprocess.check_output()` (called in the `SLiMEngine.version()` method). But the engine version should not be requested when running with --quiet.

It seems that with this PR, `args.quiet` is always None, regardless of the --quiet flag. So the existing checks for args.quiet will need to be changed, or args.quiet restored.

Reply:
Ah! OK, I'll nuke any references to ``quiet`` - there should only be ``verbose`` now.

Reply:
Should be fixed now - I changed to mocking subprocess.check_call. Seems a lot simpler?

Reply:
I don't think that's testing the right thing. The slim engine calls `Popen` to nominally run the slim executable when doing a simulation (or a dry run). The engine only uses `check_output` to get the version string.